File: rzops-web/src/lib/types/server_port_template.ts
export interface ServerPortTemplateResponse {
  id: string;
  name: string;
  protocol: string;
  port: number;
  service_name: string;
  access_scope: string | null;
  is_enabled: boolean;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface ServerPortTemplateListResponse {
  data: ServerPortTemplateResponse[];
  count: number;
}

export interface CreateServerPortTemplateRequest {
  name: string;
  protocol: string;
  port: number;
  service_name: string;
  access_scope?: string;
  is_enabled?: boolean;
  description?: string;
}

export interface UpdateServerPortTemplateRequest extends Partial<CreateServerPortTemplateRequest> {}

export interface ListServerPortTemplatesQuery {
  q?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/server_port_template.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

/// ServerPortTemplate entity — maps to cmdb_server_port_template (端口模板，用于批量录入).
#[derive(Debug, Clone)]
pub struct ServerPortTemplate {
    pub id: Uuid,
    pub name: String,
    pub protocol: String,
    pub port: i32,
    pub service_name: String,
    pub access_scope: Option<String>,
    pub is_enabled: bool,
    pub description: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
